lib/render/RayCaster.cpp

```diff
@@ -996,15 +1324,44 @@ void RayCaster::_renderTriangleStrips() const
             indexBuffer[ idx++ ] = (z + 1) * by + y; 
             indexBuffer[ idx++ ] = z * by + y;
         }
+        if( attrib1 )
+        {
+            for( unsigned int y = 0; y < by; y++ )
+            {
+                unsigned int   attribIdx       = ((z + 1) * by + y) * 4;
+                attrib1Buffer[ attribIdx     ] = int(bx) - 2; 
+                attrib1Buffer[ attribIdx + 1 ] = int(y) - 1;
+                attrib1Buffer[ attribIdx + 2 ] = int(z);
+                attrib1Buffer[ attribIdx + 3 ] = 4;
+                               attribIdx       = (z * by + y) * 4;
+                attrib1Buffer[ attribIdx     ] = int(bx) - 2; 
+                attrib1Buffer[ attribIdx + 1 ] = int(y) - 1;
+                attrib1Buffer[ attribIdx + 2 ] = int(z);
+                attrib1Buffer[ attribIdx + 3 ] = 4;
+            }
+            glBufferData( GL_ARRAY_BUFFER,      by * bz * 4 * sizeof(int),
+                          attrib1Buffer,        GL_STREAM_READ );
+            glVertexAttribIPointer( 1, 4,       GL_INT, 0, (void*)0 );
+        }
         glBufferData( GL_ELEMENT_ARRAY_BUFFER,  numOfVertices * sizeof(unsigned int),
-                      indexBuffer,              GL_STREAM_DRAW );
+                      indexBuffer,              GL_STREAM_READ );
         glDrawElements( GL_TRIANGLE_STRIP,      numOfVertices,
                         GL_UNSIGNED_INT,        (void*)0 );
     }
 
+    //
     // Render left face
+    //
```

Comment:
You are supposed to have a VertexArray for each of these faces vs reconfiguring one VAO for each draw.

Reply:
What are the tradeoffs between these two approaches?

Reply:
The approach I mentioned is the standard practice for OpenGL 3